src/server/game/Pools/PoolMgr.h

```diff
@@ -141,12 +141,11 @@ class TC_GAME_API PoolMgr
         template<typename T>
         void SpawnPool(uint32 pool_id, uint32 db_guid_or_pool_id);
 
-        uint32 max_pool_id;
-        typedef std::vector<PoolTemplateData>       PoolTemplateDataMap;
-        typedef std::vector<PoolGroup<Creature> >   PoolGroupCreatureMap;
-        typedef std::vector<PoolGroup<GameObject> > PoolGroupGameObjectMap;
-        typedef std::vector<PoolGroup<Pool> >       PoolGroupPoolMap;
-        typedef std::vector<PoolGroup<Quest> >      PoolGroupQuestMap;
+        typedef std::unordered_map<uint32, PoolTemplateData>       PoolTemplateDataMap;
+        typedef std::unordered_map<uint32, PoolGroup<Creature> >   PoolGroupCreatureMap;
```

Comment:
Since this is new code and we support c++11 there is no need to leave a space between template brackets